src/Compilers/CSharp/Portable/Syntax/CSharpSyntaxTree.cs

```diff
@@ -378,20 +380,46 @@ internal static SyntaxTree CreateWithoutClone(CSharpSyntaxNode root)
             CSharpParseOptions options = null,
             string path = "",
             CancellationToken cancellationToken = default(CancellationToken))
+        {
+            return ParseTextWithDiagnosticOptions(
+                text,
+                diagnosticOptions: null,
+                parseOptions: options,
+                path,
+                cancellationToken);
+        }
+
+        /// <summary>
+        /// Produces a syntax tree by parsing the source text with custom diagnostic options.
+        /// </summary>
+        public static SyntaxTree ParseTextWithDiagnosticOptions(
```

Comment:
>ParseTextWithDiagnosticOptions [](start = 33, length = 30)

ParseText?

Reply:
Overloading ParseText means we would create an API that has CancellationToken as the second to last parameter vs. the last parameter. I'm not sure what our policy is for this type of trade off. 

---
In reply to: [183479301](https://github.com/dotnet/roslyn/pull/25964#discussion_r183479301) [](ancestors = 183479301)

Reply:
> Overloading ParseText means we would create an API that has CancellationToken as the second to last parameter vs. the last parameter. I'm not sure what our policy is for this type of trade off. 

I am probably missing something, but I cannot think of a reason why overloading would create a problem given the current signature of this API.

---
In reply to: [183503772](https://github.com/dotnet/roslyn/pull/25964#discussion_r183503772) [](ancestors = 183503772,183479301)

Reply:
@agocke You can always suppress that warning if it's unnecessarily restrictive.  We have a lot of precedence for that in the codebase already.

Reply:
Ah yes, I see the suppression list. I'll do that.